modules/renter/contractor/editor.go

```diff
@@ -145,6 +151,21 @@ func (c *Contractor) Editor(contract modules.RenterContract) (Editor, error) {
 		}
 	}
 
+	// acquire revising lock
+	c.mu.Lock()
+	alreadyRevising := c.revising[contract.ID]
+	if alreadyRevising {
+		c.mu.Unlock()
+		return nil, errors.New("already revising that contract")
+	}
+	c.revising[contract.ID] = true
+	c.mu.Unlock()
+	releaseLock := func() {
+		c.mu.Lock()
+		delete(c.revising, contract.ID)
+		c.mu.Unlock()
+	}
```

Comment:
this violates our conventions. There's way too much code separating the acquisition and release of the lock, so it needs to be wrapped in a defer or somehow otherwise managed such that it's obvious that the resource is released correctly.